src/libslic3r/GCode.cpp

```diff
@@ -766,11 +766,15 @@ static std::vector<Vec2d> get_path_of_change_filament(const Print& print)
                 gcodegen.m_wipe.reset_path();                                           // We don't want wiping on the ramming lines.
             toolchange_gcode_str = gcodegen.set_extruder(new_extruder_id, tcr.print_z); // TODO: toolchange_z vs print_z
             if (gcodegen.config().enable_prime_tower) {
-            deretraction_str += gcodegen.writer().travel_to_z(z, "restore layer Z");
-            Vec3d position{gcodegen.writer().get_position()};
-            position.z() = z;
-            gcodegen.writer().set_position(position);
-            deretraction_str += gcodegen.unretract();
+                // ORCA: For SEMM, this move causes the nozzle to crash on the wipe tower for the final wipe layer.
+                // TODO: Requires validation whether this is an issue even with multi extruder printers
+                if(!this->m_single_extruder_multi_material){
```

Comment:
Forcing set z here is becuase we now allow users to change `position` variable in custom g-code.
e.g. in Prusa XL "change filament g-code", we have `position[2] = position[2] + 2.0;`.
It's strange why there is discrenpency in Z value in this situation, most likely in wipetower we changed z value but failed to update gcodewriter.

Reply:
Indeed…. I tried searching for it but for the life of me I couldn’t find it. Any chance you could also take a look at this one if possible? 😀 I’ll try also to find it today

Reply:
Sure, I will looking into this as well

Reply:
Thank you, really appreciate that 🙏🏻

Reply:
Change reverted as discussed :)